### DISCUSSION TITLE
How is the current state of actix 4 (with tokio 1) ?

### DISCUSSION THREAD

**Discussion by lsunsi** [actix/actix-web#2085] [Q&A]
The reason I ask is that the community at large migrated to tokio 1 and more and more I get runtime compatibility issues when trying to install or upgrade packages. I'm considering using the beta directly, but then some middlewares are not updated (which makes sense).

Since I didn't see plans or a roadmap, I'm just asking if there's any ETA or list of issues to be solved before full release so we can get our tokio 1 on, along with all the other improvements. I'm not trying to pressure anyone because I know the awesome work being done here is done voluntarily. I just wanted to know the feeling of the dev team!

Thanks a lot.

**Comment by ta3pks** [comment]
I asked a similar question over an actix cors bug but the developer was quick to close the report answering 'soon' there will be a beta release for solve unknown reason to me thru seem to be hostile when it comes to the topic of the updates

**Reply by fakeshadow** [reply]
What bug? Are you refering to this issue https://github.com/actix/actix-extras/issues/171 ?
If so it's not a bug to begin with it's working as intended. How is it hostile to close an issue clearly have misunderstanding how semver works?
As of how soon stable v4 will be released trust me if I can give a fix date I will. Do you notice there are lot of PRs no one review and lot of issue not resolved? Please understand the situation and don't spill out mis-information. No one is being hostile to you in anyway.

**Reply by robjtede** [reply]
Sorry if my quick close seemed hostile. Please understand that I get this question a lot and there is not ETA or fixed date for release. It will be released when we're happy with it and decreasing the open issue count is part of the process.

**Reply by ta3pks** [reply]
the general practice I see everywhere is to keep one issue open about such things and linking the future similar questions to that one therefore there's a place where people can follow and get notifications about the updates

**Reply by onelson** [reply]
I don't know how feasible it is, but something that could help gauge the overall progress towards the release is milestones. I notice there is a [milestone already for v4](https://github.com/actix/actix-web/milestone/6), but there aren't too many tickets linked.

If there's going to be another 3.x release slated before 4.x, it might make sense to link up outstanding issues so the 3.x line's progress can be viewed in the same way.

**Reply by fakeshadow** [reply]
> the general practice I see everywhere is to keep one issue open 

The point is your issue is not a real one. Of course cors 0.5 would not work on actix-web 4.0-betas and that's the point of versions. Not knowing how semver and version control in general work is the problem here and we can not do anything really. 

We keep every real issue open and ask people to ping for re-open when the issue re-occur.

**Reply by ta3pks** [reply]
well the point of the report is to generate a timeline for people who are like me waiting for this change to be able to update everything and so far for my project at least cors is the only blocking crate. I understand your side I hope you are understanding what I am saying. I am not trying to offend anybody or trying to get aggressive. Just this is a huge blocking thing for me and I believe there are many people like myself to have a beta would be really nice so people like myself could upgrade to it and continue business as usual

**Reply by robjtede** [reply]
as of today, cors now has a beta

**Comment by sankar-boro** [comment]
You can get an idea on how I fixed it myself from my forked repo. actix-extras is working for me on actix-beta v.4.0.0-beta.6 https://github.com/sankar-boro/actix-extras. But do not use it for production. As i have not polished them.
